providers/storage/storageDocStore.js

```diff
@@ -47,13 +47,18 @@ class AzureStorageDocStore {
       blobMetadata.extra = JSON.stringify(document._metadata.extra);
     }
     const options = { metadata: blobMetadata, contentSettings: { contentType: 'application/json' } };
-    this.service.createBlockBlobFromText(this.name, blobName, text, options, (error, result, response) => {
-      if (error) {
+    const dataStream = new Readable();
+    dataStream.push(JSON.stringify(document));
```

Comment:
Should we use a streaming JSON stringifier?

Reply:
I was contemplating about this one since the document is already fully loaded into memory. I'll add it to make this more reliable.

Reply:
Apparently, when it comes to JSON stringify streaming operation over one JSON object, the performance is far worse.